packages/di/src/decorators/inject.decorator.metadata.ts

```diff
@@ -0,0 +1,18 @@
+import { ReflectionMetadata } from '../helpers/reflection';
+import { InjectMetadata, InjectMetadataParam, InjectMetadataProperty } from './inject.decorator.types';
+
+class InjectDecoratorMetadata extends ReflectionMetadata<InjectMetadata[]> {
+  constructor() {
+    super(Symbol('inject'));
+  }
+
+  getPropertiesOnly(target: object): InjectMetadataProperty[] {
+    return (this.get(target) ?? []).filter((data) => !data.parameterIndex) as InjectMetadataProperty[];
+  }
+
+  getParamsOnly(target: object): InjectMetadataParam[] {
+    return (this.get(target) ?? []).filter((data) => !data.propertyKey) as InjectMetadataParam[];
+  }
+}
+
+export const injectDecoratorMetadata = new InjectDecoratorMetadata();
```

Comment:
I am wondering if we could put this const in pascal case as we are exporting as the instance of a not accesible class, so that it would be clear on the implementation side.